.github/workflows/update-changelog.yml

```
@@ -0,0 +1,87 @@
+          # Get all non-prerelease tags, sort, find the one before current
+          PREV_TAG=$(git tag -l 'v*' | grep -v -E '(prerelease|test|hotfix)' | sort -V | grep -B1 "^${CURRENT_TAG}$" | head -1)
+          if [ -z "$PREV_TAG" ]; then
+            # If no previous tag found, get the latest release tag
+            PREV_TAG=$(git tag -l 'v*' | grep -v -E '(prerelease|test|hotfix)' | sort -V | tail -1)
+          fi
+          echo "Auto-detected previous tag: $PREV_TAG"
+          echo "prev_tag=$PREV_TAG" >> $GITHUB_OUTPUT
+
+      - name: Check if there are commits between tags
+        id: check-commits
+        run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2086:info:2:31: Double quote to prevent globbing and word splitting

(shellcheck)